pkgs/applications/science/math/eukleides/default.nix

```
@@ -1,26 +1,57 @@
-{ lib, stdenv, fetchurl, bison, flex, texinfo, readline, texLive }:
+{ lib, stdenv, fetchurl, bison, flex, makeWrapper, texinfo, readline, texLive }:
 
-let
-  name    = "eukleides";
+lib.fix (eukleides: stdenv.mkDerivation rec {
+  pname = "eukleides";
   version = "1.5.4";
-in
-stdenv.mkDerivation {
-  name = "${name}-${version}";
 
   src = fetchurl {
-    url = "http://www.eukleides.org/files/${name}-${version}.tar.bz2";
+    url = "http://www.eukleides.org/files/${pname}-${version}.tar.bz2";
     sha256 = "0s8cyh75hdj89v6kpm3z24i48yzpkr8qf0cwxbs9ijxj1i38ki0q";
   };
 
-  buildInputs = [bison flex texinfo readline texLive];
+  nativeBuildInputs = [ bison flex texinfo makeWrapper ];
 
-  preConfigure = "sed -i 's/ginstall-info/install-info/g' doc/Makefile";
-  installPhase = "mkdir -p $out/bin ; make PREFIX=$out install";
+  buildInputs = [ readline texLive ];
+
+  preConfigure = ''
+    sed -i '/mktexlsr 2> \/dev\/null/d' Makefile
+
+    substituteInPlace doc/Makefile \
+      --replace ginstall-info install-info
+
+    substituteInPlace Config \
+      --replace '/usr/local' "$out" \
+      --replace '$(SHARE_DIR)/man "$doc/share/man" \
+      --replace '$(SHARE_DIR)/doc' "$doc/share/doc" \
+      --replace '$(SHARE_DIR)/info "$doc/share/info" \
+      --replace '$(SHARE_DIR)/texmf' "$tex"
+  '';
+
+  preInstall = ''
+    mkdir -p $out/bin
+  '';
+
+  postInstall = ''
+    # Always append a ':' to the prefix since TEXINPUTS expects
+    # one at the end to include the default path
+    wrapProgram $out/bin/euktoeps \
+      --prefix TEXINPUTS : "$tex/tex/latex/eukleides:" \
```

```suggestion
      --set-default TEXINPUTS : \
      --prefix TEXINPUTS : "$tex/tex/latex/eukleides" \
```
Does this work, by any chance? Otherwise you may get an extra `::` in the middle of `TEXINPUTS`.